### PR TITLE
feat(473): always run teardown bookends

### PR DESCRIPTION
**Pull request by minz1027**
- Run setup bookends and user steps in a single shell as what it is now
- Exit the single shell and run teardown bookends in separate shells

### CODE REVIEW

File: executor/executor.go [hunk 1]
```diff
@@ -134,40 +133,76 @@ func Run(path string, env []string, emitter screwdriver.Emitter, build screwdriv
...
+				}
+				exportString := strings.Join(exportCmd, " && ")
+
+				exec.Command(exportString)
+
+				// set teardownFlag so this block won't run again
+				teardownFlag = true
+			}
+
+			exec.Command(cmd.Cmd)
```
Comment (minz1027) [inline executor/executor.go]: This whole part should be something similar to this:
https://github.com/screwdriver-cd/launcher/blob/7be81506e9b1e34a2cab336b85fd1509bd231363/executor/executor.go#L32
We can probably just wrap in inside a separate function.

File: executor/executor.go [hunk 2]
```diff
@@ -134,40 +133,76 @@ func Run(path string, env []string, emitter screwdriver.Emitter, build screwdriv
...
 
-		if cmdErr != nil {
-			return cmdErr
+		// Set first error flag
+		if (firstError == nil) && (cmdErr != nil) {
+			firstError = cmdErr
 		}
 	}
 
 	f.Write([]byte{4}) // EOT
```
Comment (minz1027) [inline executor/executor.go]: Should move this up to replace `exec.command('exit')`

File: executor/executor.go (hunk 1 above)
Comment (d2lam) [reply executor/executor.go]: Cool! 👍

File: executor/executor.go (hunk 2 above)
Comment (d2lam) [reply executor/executor.go]: Fixed

**Pull request merged**